src/etc/inc/interfaces.inc

```diff
@@ -3006,44 +3021,76 @@ function interface_dhcpv6_prepare($interface = 'wan', $wancfg)
     $dhcp6cscript .= "\t;;\n";
     $dhcp6cscript .= "esac\n";
 
-    if (!@file_put_contents("/var/etc/dhcp6c_{$interface}_script.sh", $dhcp6cscript)) {
-        log_error("Error: cannot open dhcp6c_{$interface}_script.sh in interface_dhcpv6_configure() for writing.");
+    if (!@file_put_contents("/var/etc/dhcp6c_{$realif}_script.sh", $dhcp6cscript)) {
+        log_error("Error: cannot open dhcp6c_{$realif}_script.sh in interface_dhcpv6_configure() for writing.");
         return;
     }
 
-    chmod("/var/etc/dhcp6c_{$interface}_script.sh", 0755);
+    chmod("/var/etc/dhcp6c_{$wanif}_script.sh", 0755);
```

Comment:
all this file naming shuffling worries me. we can clean this up later according to a common understanding, but right now there's too man files that change names and variables to go with it.